src/boilerplate.rs

```diff
@@ -364,24 +365,34 @@ pub fn add_index_section(doc: &mut Spec) {
     h2_el.append(html::new_text("Index"));
     container.append(h2_el);
 
-    add_local_terms(doc, container);
-    add_external_terms(doc, container);
+    add_local_terms(doc, &container);
+    add_external_terms(doc, &container);
 }
 
+#[derive(Debug, Clone)]
 struct IndexTerm {
     url: String,
     label: String,
+    disambiguator: String,
 }
 
-fn index_items_to_node(index_items: &HashMap<String, IndexTerm>) -> NodeRef {
+fn index_items_to_node(index_items: IndexMap<String, IndexTerm>) -> NodeRef {
     let ul_el = html::new_element(
         "ul",
         btreemap! {
             "class" => "index",
         },
     );
 
-    for (link_text, index_item) in index_items {
+    for (link_text, index_item) in index_items.sorted_by(|_, index_item1, _, index_item2| {
+        if index_item1.disambiguator < index_item2.disambiguator {
```

Comment:
This whole lambda can just be written like: `index_item1.disambiguator.cmp(&index_item2.disambiguator)`

but don't you need to sort by key as well? Otherwise why do you win from this being a map?